### ISSUE TITLE
`comparison_to_empty` not triggered if comparing `Cow<'a, str>` to empty string slice

### ISSUE THREAD

**Issue by selfisekai** [rust-lang/rust-clippy#13861]
### Summary

This is effectively the same comparison, but the lint is not triggered

### Lint Name

comparison_to_empty

### Reproducer

I tried this code:

```rust
#![warn(clippy::comparison_to_empty)]

use std::borrow::Cow;

fn main() {
    let s = Cow::Borrowed("");
    if s == "" {}
}
```

I expected to see this happen:
```
warning: comparison to empty slice
 --> src/main.rs:7:8
  |
7 |     if s == "" {}
  |        ^^^^^^^ help: using `is_empty` is clearer and more explicit: `s.is_empty()`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#comparison_to_empty
```

Instead, this happened:
```
```

### Version

```text
rustc 1.83.0 (90b35a623 2024-11-26) (Alpine Linux 1.83.0-r0)
binary: rustc
commit-hash: 90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf
commit-date: 2024-11-26
host: x86_64-alpine-linux-musl
release: 1.83.0
LLVM version: 19.1.4
```
